### ISSUE TITLE
I listed a addon in ember engine's package.json under dependency and it isn't included in neither host app vendor.js file nor engine's vendor file.

### ISSUE THREAD

**Issue by swarnavallis** [ember-engines/ember-engines#645]

**Comment by villander** [comment]
You need list engines on package.json on host app, please read the guides - http://ember-engines.com/

**Comment by swarnavallis** [comment]
What if the addon is needed only in engine? It will be better if add-ons used by engine reside in engine's vendor instead of host app's vendor.

**Comment by villander** [comment]
If it is only needed in the engine, put it in the engine.

**Comment by swarnavallis** [comment]
> If it is only needed in the engine, put it in the engine.

yeah.. my issue is i have listed my engine in host app's package.json and the the addon needed by engine in engine's package.json. When i try to acess the addon inside engine i get a error "component or helper named "dialog-moal" is not found. But if i list the addon in both host app and engine 's package.json , it works fine.Why is it so?

**Comment by swarnavallis** [comment]
@villander without listing a add-on in host app, can a add-on be used in in-repo-engine by listing it under dependencies in engine's package.json ?

**Comment by villander** [comment]
> can a add-on be used in in-repo-engine by listing it under dependencies in engine's package.json ?

Yes!

**Comment by chendraayan** [comment]
@villander  @swarnavallis  I have a similar scenario. Is it mandatory to list the addon in the host app's package.json?

**Comment by swarnavallis** [comment]
@villander can you help us find where i went wrong?

**Comment by villander** [comment]
please, create a reproduction into public repo

**Comment by swarnavallis** [comment]
host' package.josn
{
devDependencies{
addon-a : 0.0.0,
addon-b : 0.0.0
}
  "ember-addon": {
    "paths": [
     "lib/engine-a"
   ]
  }
}

Engine-a package.json
dependencies
{
**addon-c:0.0.0**
}

So this is my use case. Now what will happen to addon-c. **It is only listed in engine's package.json**.Will addon-c code fall in engine-a's vendor?
@villander  @stefanpenner

**Comment by villander** [comment]
@swarnavallis it's not clear for me, I don't know how to are using `addon-c` and where.

**Comment by swarnavallis** [comment]
> @swarnavallis it's not clear for me, I don't know how to are using `addon-c` and where.

 Imagine the css or components that are only used by engine-a, but are not used by the host app. 
Now where should this components used by engine-a alone be listed?

@villander

**Comment by swarnavallis** [comment]
> > @swarnavallis it's not clear for me, I don't know how to are using `addon-c` and where.
> 
> Imagine the css or components that are only used by engine-a, but are not used by the host app.
> Now where should this components used by engine-a alone be listed?
> @villander

am using addon-c in engine-a. addon-c for example is ember-truth-helper. I need ember-truth helper inside engine-a application template and i am not using ember-truth-helper in my host app, so i listed ember-truth-helper in engine-a's package.json. But on giving npm install in project directory, it dint install the ember-truth-helper listed in engine-a's package.json.

**Comment by villander** [comment]
> Imagine the css or components that are only used by engine-a, but are not used by the host app.
Now where should this components used by engine-a alone be listed?

must be listed on `engine-a`, on package.json from `engine-a`

> am using addon-c in engine-a. addon-c for example is ember-truth-helper. I need ember-truth helper inside engine-a application template and i am not using ember-truth-helper in my host app, so i listed ember-truth-helper in engine-a's package.json. But on giving npm install in project directory, it dint install the ember-truth-helper listed in engine-a's package.json.

this must work, you're doing it correctly

**Comment by swarnavallis** [comment]
should i give npm install in engine directory or project directory? Is separate npm install required to install dependencies listed in engine's package.json

**Comment by villander** [comment]
you must run `npm install` in engine directory. that's all to do it work

**Comment by swarnavallis** [comment]
when i give npm install in engine directory, the addons that are listed in both host app and engine' package.json get inserted in both engine's and host-app's vendor file. This leads to duplication of code. The version of addon listed in both host app and engine is same. Why is this?

example:

host-app package.json
devDependencies
{ 
addon-a : "^1.1"
}

engine-a package.json
 dependencies{
addon-a : "^1.1",
addon-b: "^2.1"
}

when i run npm install on host-app package.json , the addon-a gets into host-app' vendor. Now when i run npm install in engine-a directory , addon-a gets installed again and code is inserted in engine-a vendor.js though they are of same version. 

Expected behaviour: I only want addon-b which is used by engine alone to be inserted in engine's vendor and addon-a used by both host app and engine must be in host app vendor. 
@villander

**Comment by tasha-urbancic** [comment]
> when i give npm install in engine directory, the addons that are listed in both host app and engine' package.json get inserted in both engine's and host-app's vendor file. This leads to duplication of code. The version of addon listed in both host app and engine is same. Why is this?
> 
> example:
> 
> host-app package.json
> devDependencies
> {
> addon-a : "^1.1"
> }
> 
> engine-a package.json
> dependencies{
> addon-a : "^1.1",
> addon-b: "^2.1"
> }
> 
> when i run npm install on host-app package.json , the addon-a gets into host-app' vendor. Now when i run npm install in engine-a directory , addon-a gets installed again and code is inserted in engine-a vendor.js though they are of same version.
> 
> Expected behaviour: I only want addon-b which is used by engine alone to be inserted in engine's vendor and addon-a used by both host app and engine must be in host app vendor.
> @villander

I am coming into the same use case right now. Did you ever figure this out @swarnavallis ?
